src/cli/cli.cpp

```diff
@@ -4023,12 +4060,71 @@ template <> otError Interpreter::Process<Cmd("locate")>(Arg aArgs[])
     otError      error = OT_ERROR_INVALID_ARGS;
     otIp6Address anycastAddress;
 
+    /**
+     * @cli locate
+     * @code
+     * locate
+     * Idle
+     * Done
+     * @endcode
+     * @code
+     * locate fdde:ad00:beef:0:0:ff:fe00:fc10
+     * @endcode
+     * @code
+     * locate
+     * In Progress
+     * Done
+     * @endcode
+     * @par
+     * Gets the current state (`In Progress` or `Idle`) of anycast locator.
+     * @par
+     * Available when `OPENTHREAD_CONFIG_TMF_ANYCAST_LOCATOR_ENABLE` is enabled.
+     * @par api_copy
+     * #otThreadIsAnycastLocateInProgress
+     */
     if (aArgs[0].IsEmpty())
     {
         OutputLine(otThreadIsAnycastLocateInProgress(GetInstancePtr()) ? "In Progress" : "Idle");
         ExitNow(error = OT_ERROR_NONE);
     }
 
+    /**
+     * @cli locate <anycastaddr>
+     * @code
+     * locate fdde:ad00:beef:0:0:ff:fe00:fc00
+     * fdde:ad00:beef:0:d9d3:9000:16b:d03b 0xc800
+     * Done
+     * @endcode
+     * @code
+     * srp server enable
+     * Done
+     * @endcode
+     * @code
+     * netdata show
+     * Prefixes:
+     * Routes:
+     * Services:
+     * 44970 5d c002 s c800
+     * 44970 5d c002 s cc00
+     * Done
+     * @endcode
```

Comment:
Why are srp and netdata examples in the locate one?

Reply:
They seem to be part of the example provided in the README.

Reply:
That's probably better covered in a concepts guide.  We don't do this with other commands... let's limit this to just an example of the command being documented.